hazelcast/src/test/java/com/hazelcast/query/impl/CachedQueryEntryTest.java

```diff
@@ -0,0 +1,61 @@
+package com.hazelcast.query.impl;
+
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.impl.getters.Extractors;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsSame.sameInstance;
+
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class CachedQueryEntryTest extends QueryEntryTest {
+
+    @Test
+    public void getKey_caching() {
+        QueryableEntry entry = entry("key", "value");
+
+        assertThat(entry.getKey(), sameInstance(entry.getKey()));
```

Comment:
afaik we are not using this approach in tests. We should have consistency in our tests.

Personally I don't see the added value compared to:
```
assertSame(entry.getKey(), entry.getKey())
```